src/Tools/dotnet-monitor/CollectionRules/IAction.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules
+{
+    internal interface IAction<TOptions>
```

Comment:
```suggestion
    internal interface ICollectionRuleAction<TOptions>
```

to be more specific of which system this action belongs.